scrape/scrape.go

```diff
@@ -622,6 +622,7 @@ func (sp *scrapePool) sync(targets []*Target) {
 			wg.Add(1)
 			go func(l loop) {
 				l.stop()
+				l.getCache().free()
```

Comment:
Just to learn the code a bit better, is this cache shared between scrapeLoops?

Reply:
No, it looks like `newScrapeLoop()` creates a cache per `scrapeLoop`. It does attempt to re-use an existing cache when a reload is called.